src/display/editor/annotation_editor_layer.js

```diff
@@ -95,6 +102,26 @@ class AnnotationEditorLayer {
     this.#uiManager.addLayer(this);
   }
 
+  get textLayerElements() {
+    const textLayer = this.div.parentNode
+      .getElementsByClassName("textLayer")
+      .item(0);
+    if (!textLayer) {
+      return shadow(this, "textLayerElements", null);
+    }
```

Comment:
Given how most things in the viewer are asynchronous, the textLayer could *potentially* not yet exist when this code runs or at least not have been completely rendered yet; hence this could lead to intermittent behaviour/failures.

If the textLayer has finished rendering, its last element will be a `div` with a `endOfContent`-className. Otherwise you'd have probably have to, somehow, wait for the relevant "textlayerrendered"-event before running this code.

In any case, it seems to me that this code (and probably the rest of this patch) needs to be able to properly deal with the asynchronicity of the viewer here.

Reply:
For some (possible) inspiration, please note how the `structTree`-functionality is handled in the viewer to account for the various kinds of asynchronicity involved:
https://github.com/mozilla/pdf.js/blob/a1ac1a61b75064ef274f69da4aa0da7be9a575dd/web/pdf_page_view.js#L785-L815